### PR TITLE
Add default column value support

### PR DESCRIPTION
**Pull request by trevorsibanda**
Addresses several open issues #1520  #1448 #1146 
Also provides test for #1463 

```scala
def java_sql_Timestamp = column[java.sql.Timestamp]("java_sql_Timestamp" , O.Default("2016-06-08 19:34:34") )
def Option_java_sql_Timestamp = column[Option[java.sql.Timestamp]]("Option_java_sql_Timestamp" , O.Default( Some("CURRENT_TIMESTAMP") ) )
```

### CODE REVIEW

File: slick/src/main/scala/slick/relational/RelationalProfile.scala [hunk 1]
```diff
@@ -109,6 +109,8 @@ trait RelationalTableComponent { self: RelationalProfile =>
   trait ColumnOptions {
     val PrimaryKey = ColumnOption.PrimaryKey
     def Default[T](defaultValue: T) = RelationalProfile.ColumnOption.Default[T](defaultValue)
+    def Default[T](defaultValue: String) = RelationalProfile.ColumnOption.Default[T](defaultValue)
+    def Default[T](defaultValue: Option[String]) = RelationalProfile.ColumnOption.Default[T](defaultValue)
```
Comment (cvogt) [inline slick/src/main/scala/slick/relational/RelationalProfile.scala]: why are these overloads needed?

File: slick/src/main/scala/slick/relational/RelationalProfile.scala [hunk 2]
```diff
@@ -90,7 +90,7 @@ object RelationalProfile {
   /** Extra column options for RelationalProfile */
   object ColumnOption {
     /** Default value for the column. Needs to wrap an Option for nullable Columns. */
-    case class Default[T](val defaultValue: T) extends ColumnOption[T]
+    case class Default[T](val defaultValue: Any) extends ColumnOption[T]
```
Comment (cvogt) [inline slick/src/main/scala/slick/relational/RelationalProfile.scala]: Why Any?

File: slick/src/main/scala/slick/jdbc/JdbcTypesComponent.scala [hunk 3]
```diff
@@ -86,7 +88,7 @@ trait JdbcTypesComponent extends RelationalTypesComponent { self: JdbcProfile =>
   abstract class DriverJdbcType[@specialized T](implicit val classTag: ClassTag[T]) extends JdbcType[T] {
     def scalaType = ScalaBaseType[T]
     def sqlTypeName(sym: Option[FieldSymbol]): String = self.defaultSqlTypeName(this, sym)
-    def valueToSQLLiteral(value: T) =
+    override def valueToSQLLiteral[A](value: A) =
```
Comment (cvogt) [inline slick/src/main/scala/slick/jdbc/JdbcTypesComponent.scala]: What was the problem with T here?

File: slick/src/main/scala/slick/jdbc/JdbcType.scala [hunk 4]
```diff
@@ -34,7 +34,8 @@ trait JdbcType[@specialized(Byte, Short, Int, Long, Char, Float, Double, Boolean
 
   /** Convert a value to a SQL literal.
     * This should throw a `SlickException` if `hasLiteralForm` is false. */
-  def valueToSQLLiteral(value: T): String
+  def valueToSQLLiteral[A](value: A): String
```
Comment (cvogt) [inline slick/src/main/scala/slick/jdbc/JdbcType.scala]: What was the problem with T here?

**Comment by cvogt** [comment]
@trevorsibanda test fail here. We can try to pair code this weekend to finish this and a few other things up. Maybe Saturday afternoon/evening?

**Comment by trevorsibanda** [comment]
@cvogt That would be great. Saturday afternoon/evening is great

**Comment by cvogt-gitter** [comment]
awesome, let's aim for that

**Comment by cvogt** [comment]
superseeded by #1533

**Pull request closed without merging**